### PR TITLE
Auto populate list of programs

### PR DESCRIPTION
**Pull request by jturney**
The model currently in QCEngine requires the coder to add their new Python file and then modify two other locations to enable it.  This PR allows the coder to simply add their new Python file to the proper location and have a similarly named function inside of it and that's it.

### CODE REVIEW

**Comment by loriab** [comment]
I worry a little that this might catch strays with all the `psi4.py`, `core.so`, `plugin.so`s running around.

**Comment by dgasmith** [comment]
What about a `qcengine.programs.register_program(alias, function)` kind of thing where we have a global dict in the `programs` folder that we can add to via the above. 

Sorry, I didn't get back to this earlier. I had forgotten to watch this repo.

**Comment by jturney** [comment]
There shouldn't be a problem catching strays as it only imports files from that directory. Doing `qcengine.programs.register_program(alias, function)` may be a good idea if we ever need multiple entry points in a file. I'll work on changing over to that.

**Comment by dgasmith** [comment]
Let me know if you need help getting this in.

**Comment by dgasmith** [comment]
I will likely do this in the next few days, let me know if I should not!

**Comment by jturney** [comment]
I'm working on this right now!

File: qcengine/compute.py [hunk 1]
```diff
@@ -36,15 +36,12 @@ def compute(input_data, program, raise_error=False, capture_output=True):
...
-            output_data = programs.run_rdkit(input_data)
-        elif program == "torchani":
-            output_data = programs.run_torchani(input_data)
-        else:
+        try:
+            output_data = programs.get_program(program)(input_data)
+        except Exception as e:
             output_data["success"] = False
-            output_data["error_message"] = "QCEngine Call Error:\nProgram {} not understood".format(program)
+            output_data["error_message"] = "QCEngine Call Error:\nProgram {} not understood.\nError Message: {}".format(
```
Comment (dgasmith) [inline qcengine/compute.py]: As a note the wrapper handles Exception capturing. This try/except should explicitly be for key errors.

Comment (jturney) [reply qcengine/compute.py]: I'll add it and resolve the conflict shortly.

**Comment by dgasmith** [comment]
Merging, thanks for the PR!

**Pull request merged**